numalogic/models/autoencoder/variants/vanilla.py

```diff
@@ -182,25 +182,25 @@ def __init__(
 
     @staticmethod
     def init_weights(m: nn.Module) -> None:
-        r"""Initiate parameters in the transformer model."""
+        """Initialize the parameters in the model."""
         if type(m) == nn.Linear:
             nn.init.xavier_normal_(m.weight)
 
     def forward(self, batch: Tensor) -> tuple[Tensor, Tensor]:
-        batch = batch.view(-1, self.n_features, self.seq_len)
+        batch = torch.swapdims(batch, 1, 2)
```

Comment:
Earlier, when `batch.view` was applied, each feature vector sent to each channel carries datapoints interleaved from different metrics (such as error%, cpu, memory). Now, with `torch.swapdims`, are we merely sending datapoints of different metrics to different channels?

If the answer is 'yes', we would be detecting pattern mismatches for each metrics independent of the others.

Reply:
I think you are misunderstanding the requirement here. We are moving/swapping the dims because the convolutional operation requires the tensor to be aligned in a particular dimensional consistency. https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html
Regarding the dense layers, the contraction on the dimensions is happening on the sequence length dim, not on the feature dim.